src/lava/lib/optimization/problems/problems.py

```diff
@@ -187,90 +196,89 @@ class QP:
-            self._constraint_biases = constraint_biases
-        else:
-            self._constraint_hyperplanes = None
-            self._constraint_biases = None
-
-        if constraint_hyperplanes_eq is not None:
-            self._constraint_hyperplanes_eq = constraint_hyperplanes_eq
-            self._constraint_biases_eq = constraint_biases_eq
-
-        if constraint_hyperplanes_eq is not None:
-            constraint_hyperplanes_eq_new = np.vstack(
-                (constraint_hyperplanes_eq, -constraint_hyperplanes_eq)
-            )
-            constraint_biases_eq_new = np.vstack(
-                (constraint_biases_eq, -constraint_biases_eq)
-            )
-            if constraint_hyperplanes is not None:
-                self._constraint_hyperplanes = np.vstack(
-                    (
-                        self._constraint_hyperplanes,
-                        constraint_hyperplanes_eq_new,
-                    )
-                )
-                self._constraint_biases = np.vstack(
-                    (self._constraint_biases, constraint_biases_eq_new)
-                )
-            else:
-                self._constraint_hyperplanes = constraint_hyperplanes_eq_new
-                self._constraint_biases = constraint_biases_eq_new
+        super().__init__()
+        self.c_variables = ContinuousVariables(hessian.shape[0])
+        self.q_cost = Cost(linear_offset, hessian)
+        self._constraints.arithmetic = ArithmeticConstraints(
+            ineq=[constraint_biases, constraint_hyperplanes],
+            eq=[constraint_biases_eq, constraint_hyperplanes_eq],
+        )
+
+    @property
+    def variables(self):
+        return self.c_variables
+
+    @property
+    def cost(self):
+        return self.q_cost
+
+    @property
+    def constraints(self):
+        return self._constraints.arithmetic
 
     @property
     def get_hessian(self) -> np.ndarray:
-        return self._hessian
+        return self.cost.get_coefficient(order=2)
 
     @property
     def get_linear_offset(self) -> np.ndarray:
-        return self._linear_offset
+        return self.cost.get_coefficient(order=1)
 
     @property
     def get_constraint_hyperplanes(self) -> np.ndarray:
-        return self._constraint_hyperplanes
+        return self.constraints.inequality.get_coefficient(order=2)
 
     @property
     def get_constraint_biases(self) -> np.ndarray:
-        return self._constraint_biases
+        return self.constraints.inequality.get_coefficient(order=1)
 
     @property
     def num_variables(self) -> int:
-        return len(self._linear_offset)
+        return self.variables.num_variables
+
+
+class LP(OptimizationProblem):
+    def __init__(
+        self,
+        c,
+        A_ub=None,
+        b_ub=None,
+        A_eq=None,
+        b_eq=None,
+        bounds=None,
+        x0=None,
+    ):
+        """Example adopting the interface from scipy.optimize.linprog.

```

Comment:
This docstring is not really clear to me.
In addition, I'm unsure if we should mention that we took inspiration from another package. IP issues?